Minimal 'CoreDataPlace' Widget Example

rsimon opened the pull request:
## In this PR

This PR adds a new component to the `geospatial` package: `<CoreDataPlace>`. This component retrieves a Place resource from the Core Data API, and renders it's GeoJSON geometry on a mapLibre map.

CoreDataPlace takes the following input props:

- a mapLibre basemap style 
- the URI to the Place resource on the Core Data API
- optionally: mapLibre style definitions for `fillStyle`,  `strokeStyle` and `pointStyle`

Note: for now, this component is minimal, and supposed to serve as an example only!

<img width="824" alt="Bildschirmfoto 2024-01-27 um 09 13 47" src="https://github.com/performant-software/react-components/assets/470971/80a121b1-afce-4213-8ac5-b075f7f4e20a">

dleadbetter commented on packages/storybook/package.json:
```diff
@@ -14,6 +14,8 @@
     "@babel/preset-flow": "^7.22.5",
     "@babel/preset-react": "^7.22.5",
     "@faker-js/faker": "^8.0.2",
+    "@peripleo/maplibre": "^0.3.1",
```
Should we also add these dependencies to the `geospatial` package?

rsimon replied:
Of course - done!

dleadbetter reviewed (commented):
This looks great! I might actually poach a couple of the components/styles for a backend Core Data feature I'm working on. A couple of questions/notes:
- We're using [FlowJS](https://flow.org/) as a static type checker and ESLint for syntax. Could we make sure the code style is conforming to the rules we've setup?
- Would it make sense to export the `CoreDataPlaceLayer` component? And add what is essentially the `CoreDataPlace` component (Peripleo, Map, etc) to the `.stories` file?

rsimon commented:
I made a few amendments:
- Added @peripleo dependencies to `geospatial`.
- Changed from `interface` to `type`.

I made a few tweaks to adapt the code style. I also ran `yarn flow` and that didn't report any errors. But not sure if I'm using it wrong...

Regarding the `interface` vs. `type` issue: that definitely was one of the issues. I'm one step further, but it's still not working. Storybook complains about @peripleo not exporting `Feature` and `FeatureCollection` for example. Both of which are `interfaces`. (In fact the log output lists all of the exports - and none of the `interface`s are there.) 

Is there anything we can change about the configuration to make Storybook recognize interfaces? Otherwise, I'll change all the interfaces defined in @peripleo to types.

dleadbetter commented on packages/geospatial/package.json:
```diff
@@ -22,6 +22,8 @@
   },
   "devDependencies": {
     "@performant-software/webpack-config": "^1.0.0",
+    "@peripleo/maplibre": "^0.3.2",
```
Should these be in `dependencies` instead of `devDependencies`?

rsimon replied:
Done!

dleadbetter commented on packages/geospatial/src/components/CoreDataPlace.js:
```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from 'react';
```
I think if you add the `// @flow` annotation to the top of this file, it will like the syntax a lot more. There's also a few ESLint errors on the formatting, but less concerned about those.

rsimon replied:
Done, but even with Flow enabled, I'm still getting red squiggles under every TypeScript statement, and Flow shows all the TS stuff as "uncovered area". I'll enquire with Camden.

rsimon commented:
(Partial) success at last! Thanks for the hints - yes, those deps should have been under `dependencies` not `devDependencies` of course. Sorry for the mishap. (I shouldn't have copy-and-pasted those into the package.json after all...)

After that, the component showed up in storybook. A few things were still missing, which I added:

- Import of essential CSS styles for Peripleo + minimal CSS config needed for the component
- Changed the demo map style to one that works without a MapTiler API key

<img width="1074" alt="Bildschirmfoto 2024-01-31 um 07 53 24" src="https://github.com/performant-software/react-components/assets/470971/f8bd1f3e-30ee-46a0-a53f-cf8bdb684647">

I also added the `// @flow` directive. This did seem to change at least something. (The Flow status indicator in VS code now shows that the file is begin covered.) But otherwise had no effect on syntax highlighting. Still getting errors shown about TS not being allowed in `.js` files. I'll enquire with Camden.

dleadbetter commented:
Closing in favor of #252.
